ml-agents/mlagents/trainers/stats.py

```diff
@@ -304,6 +343,39 @@ def __init__(self, category: str):
     def add_writer(writer: StatsWriter) -> None:
         StatsReporter.writers.append(writer)
 
+    @staticmethod
+    def load_state(path: str) -> None:
+        """
+        Load a JSON file that contains saved state.
+        :param path: Path to the JSON file containing the state.
+        """
+        try:
+            with open(path, "r") as f:
+                loaded_dict = json.load(f)
+            # Compare the metadata
+            _metadata = loaded_dict[StatsPropertyType.STATS_METADATA.value]
+            StatsMetaData.from_dict(_metadata).check_compatibility(StatsMetaData())
+            # Update saved state.
+            StatsReporter.saved_state.update(loaded_dict)
+        except FileNotFoundError:
+            pass
```

Comment:
Should there be a warning here, or is it expected that this won't be there most of the time?

Reply:
Added a warning. This should only happen if the user is loading from an older version of ML-Agents that did not save out the `training_status.json` file, or (as Vince mentioned below) the file was not saved out due to a crash.